src/main/java/myapp/storage/Storage.java

```diff
@@ -0,0 +1,99 @@
+package myapp.storage;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import myapp.task.*;
+import myapp.exception.RubyException;
+
+/**
+ * The {@code Storage} class handles reading from and writing to a file that stores tasks.
+ * It allows tasks to be loaded into the application at startup and saved back to the file on updates.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs a {@code Storage} object with the specified file path.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file specified by the file path.
+     * If the file does not exist, an empty list of tasks is returned.
+     *
+     * @return A list of tasks read from the file.
+     * @throws IOException    If an I/O error occurs while reading from the file.
+     * @throws RubyException  If the file contains an invalid task type or format.
+     */
+    public List<Task> load() throws IOException, RubyException {
+        List<Task> tasks = new ArrayList<>();
+        Path path = Paths.get(filePath);
+        if (!Files.exists(path)) {
+            return tasks;
+        }
+
+        List<String> lines = Files.readAllLines(path);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        for (String line : lines) {
+            String[] parts = line.split(" \\| ");
+            Task task;
+            switch (parts[0]) {
```

Comment:
The switch statement could be trimmed a little by using the extended switch expression syntax (which you used in  lines 38-46 of Parser.java).